app/rankers/algorithms.py

```diff
@@ -113,12 +113,57 @@ def thompson_sampling(
 
 
 def personalize_topic_slates(input_slate_configs: List['SlateConfigModel'],
+                             personalized_topics: PersonalizedTopicList,
+                             topic_limit: Optional[int] = 1) -> List['SlateConfigModel']:
+    """
+    This routine takes a list of slates as input in which must include slates with an associated curator topic
+    label.  It uses the topic_profile that is supplied by RecIt to re-rank the slates according to affinity
+    with items in the user's list.
+    This version allows non-topic slates within the lineup.  These are left in order in the output configs
+    list.  Personalizable (topic) slates are re-ordered using their initial slots in the config lineup.
+    If the topic_limit parameter is included this will determine the number of topic slates that
+    remain in the output config list.
+    :param input_slate_configs: SlateConfigModel list that includes slates with curatorTopicLabels
+    :param personalized_topics: response from RecIt listing topics ordered by affinity to user
+    :param topic_limit: desired number of topics to return, if this is set the number of slates returned is truncated.
+                        otherwise all personalized topics among the input slate configs are returned
+    :return: SlateLineupExperimentModel with reordered slates
+    """
+    topic_to_score_map = {t.curator_topic_label: t.score for t in personalized_topics.curator_topics}
+    # filter non-topic slates
+    personalizable_configs = list(filter(lambda s: s.curator_topic_label in topic_to_score_map, input_slate_configs))
+    print(personalizable_configs)
```

Comment:
Metaflow captures print output and identifies the step so it is easy to track down. We haven't been using the logging statements as much in the flows since they don't seem to provide any useful additional information.